io_u.h

```diff
@@ -81,7 +81,7 @@ struct io_u {
 	struct io_piece *ipo;
 
 	unsigned long long resid;
-	unsigned int error;
+	int error;
```

Comment:
Changes like this HAS to be done very carefully, and not just because some static checker has a blurb about it. The error field is positive values, did you consider that perhaps it's that one use case that may be broken? That would be worthwhile fixing up.